### PR TITLE
[#32] Avatar component

### LINKED ISSUE #32: Avatar
## Tasks
- [ ] Avatar Component - 2MD

### PR DESCRIPTION
**Pull request by daisy142**
**What this PR does / why we need it**:

- Add avatar component from ParkUI
- Update and custom styles for avatar component

**Special notes for your reviewer**:

**Which issue(s) this PR fixes**:

Fixes [#32](https://github.com/yorkie-team/yorkie-ui-poc/issues/32)

**Screenshoots**:
Screenshot 2024-01-08 at 12 04 50

**Checklist**:
- [ ] Added relevant tests or not required
- [x] Didn't break anything

### CODE REVIEW

File: src/components/ui/avatar/Avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { Avatar as ArkAvatar, type AvatarProps as ArkAvatarProps } from '@ark-ui/react/avatar';
+import { forwardRef } from 'react';
+import { avatar, type AvatarVariantProps } from '@/styled/recipes';
+
+export interface AvatarProps extends ArkAvatarProps, AvatarVariantProps {
```
Comment (vitran12) [inline src/components/ui/avatar/Avatar.tsx]: You should new file name: types.ts and move interface and type into it.

File: src/components/ui/avatar/Avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  name?: string;
```
Comment (vitran12) [inline src/components/ui/avatar/Avatar.tsx]: You should note comment for name property. example: 

 /** name is plalala */

File: src/components/ui/avatar/Avatar.tsx [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+  src?: string;
+}
+
+export const Avatar = forwardRef<HTMLDivElement, AvatarProps>((props, ref) => {
```
Comment (vitran12) [inline src/components/ui/avatar/Avatar.tsx:14]: You should write a description for the function and param of us. Example:
```
/**
 * @description This function render style for Avatar.
 */
```

File: src/components/ui/avatar/Avatar.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+Avatar.displayName = 'Avatar';
+
+const UserIcon = () => (
+  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24" fill="none" stroke="currentColor" strokeWidth="2">
+    <path d="M19 21v-2a4 4 0 0 0-4-4H9a4 4 0 0 0-4 4v2" />
+    <circle cx="12" cy="7" r="4" />
+  </svg>
+);
+
+const getInitials = (name = '') =>
```
Comment (vitran12) [inline src/components/ui/avatar/Avatar.tsx]: What is a util?
If this is a util, you can move it into util file

File: src/components/ui/avatar/Avatar.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+    <ArkAvatar.Root ref={ref} className={styles.root} {...rootProps}>
+      <ArkAvatar.Fallback className={styles.fallback}>{getInitials(name) || <UserIcon />}</ArkAvatar.Fallback>
+      <ArkAvatar.Image className={styles.image} src={src} alt={name} />
+    </ArkAvatar.Root>
+  );
+});
+
+Avatar.displayName = 'Avatar';
+
+const UserIcon = () => (
```
Comment (vitran12) [inline src/components/ui/avatar/Avatar.tsx]: You should we will create 1 file manage Icons.
Example

Import {UserIcon} from 'icons'.

**Review by vitran12 — COMMENTED** [review]
Thank you for your hard working.
I left a few comment.

**Comment by daisy142** [comment]
> Thank you for your hard working. I left a few comment.

I updated. Please help me check it before I update others. Thanks a lot @vitran12

**Review by vitran12 — APPROVED** [review]

**Pull request merged**